discord/commands/delete/index.js

```diff
@@ -0,0 +1,15 @@
+const { Command } = require("nodejsdiscordbot");
+
+module.exports = class DeletePost extends Command {
+    constructor() {
+        super(
+            "delete",
+            ["id"],
+            ["ADMINISTRATOR"]
+            );
+    }
+
+    async _run(ctx, args) {
+        return ctx.channel.send("Deleting post...");
+    }
+}
```

Comment:
This is something I'd like to see become a single function

Reply:
The way I coded the NodeJSDiscord bot package means that each command has to be its own class.
I could rework the library to just accept functions if necessary.

Reply:
I think libraries should be built so the installed tool can use it however they want. You should never dictate how someone else writes their code (ironic coming from this comment). But you should make the consuming tool have the lowest bar of entry possible

Reply:
I built it that way to allow the bot to access more than just the ctx and args passed from discord.js. With functions there would be no signal passed for the main class, KnicksRedditBot, to interact with and then delete the actual post.

I don't think its dictating how you write the code as you could easily add a different method to the class to interact with discord messages. It's more of a guideline.